### ISSUE TITLE
Frozen Gradio Image Update

### ISSUE THREAD

**Issue by smamindl** [gradio-app/gradio#5145]
### Describe the bug

Hi @abidlabs @freddyaboulton 

I have the following code snippet which is taken from your demo examples. The code runs fine locally. However, when we run it in a GCP or OCI server with public IP, it freezes after a couple of seconds and then we need to stop and start again. could you please help me with this? Thanks.

```
import gradio as gr
import numpy as np
import time

# define core fn, which returns a generator {steps} times before returning the image
def fake_diffusion(steps):
    for _ in range(steps):
        time.sleep(0.1)
        image = np.random.random((600, 600, 3))
        yield image
    image = "https://gradio-builds.s3.amazonaws.com/diffusion_image/cute_dog.jpg"
    yield image


demo = gr.Interface(fake_diffusion, inputs=gr.Slider(1, 10000, 5000), outputs="image")

# define queue - required for generators
demo.queue()

demo.launch()

```

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```
import gradio as gr
import numpy as np
import time

# define core fn, which returns a generator {steps} times before returning the image
def fake_diffusion(steps):
    for _ in range(steps):
        time.sleep(0.1)
        image = np.random.random((600, 600, 3))
        yield image
    image = "https://gradio-builds.s3.amazonaws.com/diffusion_image/cute_dog.jpg"
    yield image


demo = gr.Interface(fake_diffusion, inputs=gr.Slider(1, 10000, 5000), outputs="image")

# define queue - required for generators
demo.queue()

demo.launch()

```

### System Info

```shell
Running this on OCI/GCP server.
```

### Severity

Blocking usage of gradio

**Comment by smamindl** [comment]
Hi Team, @aliabd , @freddyaboulton @aliabid94 @gradio-pr-bot 

Any update on this? This is blocking my usage of Gradio. It would be great If you can take a look at this sooner. Thanks.

**Comment by smamindl** [comment]
Hi @freddyaboulton ,

I agree that the issue is dupe and closed. Apologies for creating 2 issues for the same problem, I could not find an option to tag it as bug after posting the issue and hence created another issue with bug tag. However, the issue is not resolved.
@abidlabs tagged it as cloud few minutes back and this issue still needs to be addressed. 

Thanks.

**Comment by smamindl** [comment]
Hi Team, 

Is this is known issue? Can someone please clarify? Thanks.

**Comment by freddyaboulton** [comment]
Hi @smamindl - gradio applications with the queue enabled connect via a websocket. Is it possible your GCP instance does not allow websocket connections? The websocket connection is created in the `/queue/join` route.

**Comment by smamindl** [comment]
> Hi @smamindl - gradio applications with the queue enabled connect via a websocket. Is it possible your GCP instance does not allow websocket connections? The websocket connection is created in the `/queue/join` route.

Hi @freddyaboulton
websocket is open on GCP. Here are the screenshots of the network logs:
<img width="1436" alt="Screen Shot 2023-08-10 at 9 53 05 AM" src="https://github.com/gradio-app/gradio/assets/106691906/c2d282fe-ceef-45e4-911f-05a6605ee0df">
<img width="1439" alt="Screen Shot 2023-08-10 at 9 53 18 AM" src="https://github.com/gradio-app/gradio/assets/106691906/6108b6c5-dd8f-445e-9ac5-76dcb8eb0f41">



It listens and then stops at some point.

**Comment by smamindl** [comment]
Hi @abidlabs @freddyaboulton 

After debugging this case, I found the issue is coming from the following snippet inside https://github.com/gradio-app/gradio/blob/main/gradio/queueing.py 

```
 async def send_message(self, event, data: dict, timeout: float | int = 1) -> bool:
        try:
            await asyncio.wait_for(
                event.websocket.send_json(data=data), timeout=timeout
            )
            return True
        except Exception:
            await self.clean_event(event)
            return False
```
Here timeout is hardcoded as 1 sec and in the case of cloud platforms due to the network traffic the chance that timeout happens is very high and application goes to invalid mode where send message is terminated while web socket is still open.
So, I increased the timeout and the chance of getting stuck reduced. I think perfect solution would be to figure out how to trigger func send_message if the websocket is still open. Simple workaround is to program this timeout using config file because currently I have to patch this file inside docker.
